### ISSUE TITLE
Policies we can apply several times for a fleet overwrite each other

### ISSUE THREAD

**Issue by btry** [flyve-mdm/glpi-plugin#302]
#### Observed Results:

If we deploy several apps to a fleet, all MQTT messages will share the same topic. The last app will overwrite all others.

Same for file deployment, and in general all policies without unicity.
  
#### Expected behavior:
MQTT messages mst be persisted and have a different topic. Let's create a sub level in the MQTT tree for them.

**Comment by btry** [comment]
@rafaelje 

Here is my proposal.

As far as I understand, your implementation of the agent is not sensitive to the MQTT topic when it receives a policy applied on a fleet.

We already agreed to append an ID as a sub topic of the current implementation fro the non unnique policies (app / file deployment and removal).

We may go a bit further by defining the ID being the Task ID, and append it for all policies, no matter the unicity of a policy. Doing this would allow us to remove the task ID in the MQTT message (in  the JSON string), while still allowing you to retrieve this ID and use it for feedback (to be implemetned)

Here is the proposal 

```php
$fleet->notify("$topic/Policy/$policyName/Task/$taskId", $encodedMessage, 0, 1);
```
$topic = the fleet topic
$policy is the symbol of the policy
$tasiId is the ID of the  task, the one you already receive in the JSON

This way we also reduce the verall size of a MQTT packet (approximatively 7 bytes)

**Comment by rafaelje** [comment]
You are right the agent is not sensitive to the MQTT topic name but if we add the Task/$taskId on the topic and remove from the JSON this change affect the actual behavior of the agent.

I think we need to avoid send arguments or data on the topic to prevent split the topic to get the task id, IMO in this case, we can use the Task id on the topic to isolate the message but keeping the taskId on the JSON message too

**Comment by btry** [comment]
One of the best application of MQTT is when a device sends data about sensors.

Imagine something having dozens or hundreds of sensors. All sensors are organized in a tree hierarchy. This builds a MQTT topic. I think this is acceptable to have some data in the MQTT topic (the name of the sensor for example). Sending a name or an ID in a topic seems a good practice. It is easy to split with a slash as separator.

Anyway, I don't plan to implement right now the ideas I described. I want to focus on a fix, and I think this would be better to have the same topic structure for everything. Less logic both for you and me (and less bugs ! )

If you agree with the sample I submitted above, I can implement it. I will not remove the task id in the JSON, and I'll not ask you to get the task ID from the topic for now. I suggest we open a new issue about this, and plan it for 3.0.
